plugins/org.eclipse.sirius/src/org/eclipse/sirius/business/api/image/ImageManager.java

```diff
@@ -36,7 +49,89 @@ public interface ImageManager {
     String HTML_IMAGE_PATH_PATTERN = "<img.*?src=\"(.*?)\".*?/>"; //$NON-NLS-1$
 
     /**
-     * Create and serialize a file from a Base64 encoding image. It returns the path that will be used to get this file.
+     * Timestamp format for image filenames generated.
+     */
+    String IMAGE_NAME_FORMAT = "yyyyMMdd_HHmmss_SSSSSS"; //$NON-NLS-1$
+
+    /**
+     * This class store the function that creates the file and its file name. Result of {@link #getCreateFileFunc}
+     * 
+     * @author scosta
+     *
+     */
+    final class CreateFileFunc {
+
+        private Supplier<Optional<CoreException>> createFileFunc;
```

Comment:
I was wondering if we could not just having the Supplier<Optional<String>> which could return the computed fileName (passed in its constructor maybe) when the get() is called